src/tools/miri/tests/pass-dep/shims/libc-fs-with-isolation.rs

```diff
@@ -22,7 +22,8 @@ fn main() {
     }
 
     // test `stat`
-    assert_eq!(fs::metadata("foo.txt").unwrap_err().kind(), ErrorKind::PermissionDenied);
+    let err = fs::metadata("foo.txt").unwrap_err();
+    assert_eq!(err.kind(), ErrorKind::PermissionDenied);
     // check that it is the right kind of `PermissionDenied`
-    assert_eq!(Error::last_os_error().raw_os_error(), Some(libc::EACCES));
+    assert_eq!(err.raw_os_error(), Some(libc::EACCES));
```

Comment:
I'm fine with this change. However, the `raw_os_error` docs do indicate that this does return the error from the previous stdlib call, so isn't it strange that this changed?

Reply:
`errno` is changed to `EFAULT` because the `statx` state is checked after the initial call returned an error. Since `EFAULT` indicates that `statx` is available, the error returned is the one by the first `statx` call, which has a different error code (like `EACCES`).